botbot_plugins/plugins/jira.py

```diff
@@ -8,55 +9,79 @@
 
 
 class Config(config.BaseConfig):
-    jira_url = config.Field(help_text="Jira Link, eg: 'https://tickets.metabrainz.org'")
-    rest_api_suffix = config.Field(help_text="Suffix for the Jira REST API, eg: 'rest/api/2/project'", default="rest/api/2/project")
+    jira_url = config.Field(help_text="JIRA Link, eg: 'https://tickets.metabrainz.org'")
+    rest_api_suffix = config.Field(help_text="Suffix for the JIRA REST API, eg: 'rest/api/2/project'", default="rest/api/2/project")
     bot_name = config.Field(help_text="Name of your bot, eg: BrainzBot")
 
 class Plugin(BasePlugin):
     """
-    Jira issue lookup
+    JIRA issue lookup
 
-    Returns the description of a Jira Issue
+    Returns the description of a JIRA Issue
 
         jira:{{projectname}}-{{issuenumber}}
     """
     config_class = Config
     
-    @listens_to_all(ur'(?:.*)\b(?P<project>\w+)-(?P<issue>\d+)\b(?:.*)')
-    def issue_lookup(self, line, project, issue):
-        """Lookup a specified jira issue
-
-            Usage:
-                Just mention the issue by its {{ISSUENAME}}
-                Eg:
-                    Can you please checkup on PROJECT-123
+    @listens_to_all(ur'(?:.*)\b(\w+-\d+)\b(?:.*)')
+    def issue_lookup(self, line):
         """
+        Lookup a specified JIRA issue
 
-        api_url = urljoin(self.config['jira_url'], self.config['rest_api_suffix'])
-        projects = json.loads(self.retrieve('projects'))
-        if project.upper() in projects and line.user != self.config['bot_name']:
-            issue_url = urljoin(api_url,"issue/{}-{}".format(project.upper(),(issue)))
-            response = requests.get(issue_url)
-            if response.status_code == 200:
-                response_text = json.loads(response.text)
-                name = response_text['key']
-                desc = response_text['fields']['summary']
-                return_url = urljoin(self.config['jira_url'],"projects/{}/issues/{}".format(project,name))
-                return "{}: {}\n{}".format(name,desc,return_url)
+        Usage:
+            Just mention the issue by its {{ISSUENAME}}
+            Eg:
+                Can you please checkup on PROJECT-123
+        """
+
+        if line.user not in (self.config['bot_name'],'github'):
+            api_url = urljoin(self.config['jira_url'], self.config['rest_api_suffix'])
+            projects = json.loads(self.retrieve('projects'))
+
+            queries = re.findall(r"\w+-\d+", line.text)
```

Comment:
you could match only upper-case jira keys by using `[A-Z]` instead of `\w`. I would check with someone who maintains our Jira instance that we will only ever have upper-case project keys.